spec/unit/shell/shell_session_spec.rb

```diff
@@ -180,68 +154,65 @@ def loading_complete
-    Shell::Extensions.extend_context_recipe(@recipe)
-  end
-
-  after do
-    Chef::Config[:shell_solo] = nil
+    Chef::Config[:solo_legacy_shell] = true
+    session.node = node
+    session.json_configuration = json_attribs
+    session.run_context = run_context
+    session.recipe = recipe
+    Shell::Extensions.extend_context_recipe(recipe)
   end
 
   it "returns a collection based on it's compilation object and the extra recipe provided by chef-shell" do
-    allow(@session).to receive(:node_built?).and_return(true)
+    allow(session).to receive(:node_built?).and_return(true)
     kitteh = Chef::Resource::Cat.new("keyboard")
-    @recipe.run_context.resource_collection << kitteh
-    expect(@session.resource_collection.include?(kitteh)).to be true
+    recipe.run_context.resource_collection << kitteh
+    expect(session.resource_collection.include?(kitteh)).to be true
   end
 
   it "returns definitions from its compilation object" do
-    expect(@session.definitions).to eq(@run_context.definitions)
+    expect(session.definitions).to eq(run_context.definitions)
   end
 
   it "keeps json attribs and passes them to the node for consumption" do
-    @session.node_attributes = { "besnard_lakes" => "are_the_dark_horse" }
-    expect(@session.node["besnard_lakes"]).to eq("are_the_dark_horse")
+    session.node_attributes = { "besnard_lakes" => "are_the_dark_horse" }
+    expect(session.node["besnard_lakes"]).to eq("are_the_dark_horse")
     # pending "1) keep attribs in an ivar 2) pass them to the node 3) feed them to the node on reset"
   end
 
   it "generates its resource collection from the compiled cookbooks and the ad hoc recipe" do
-    allow(@session).to receive(:node_built?).and_return(true)
+    allow(session).to receive(:node_built?).and_return(true)
     kitteh_cat = Chef::Resource::Cat.new("kitteh")
-    @run_context.resource_collection << kitteh_cat
+    run_context.resource_collection << kitteh_cat
     keyboard_cat = Chef::Resource::Cat.new("keyboard_cat")
-    @recipe.run_context.resource_collection << keyboard_cat
-    # @session.rebuild_collection
-    expect(@session.resource_collection.include?(kitteh_cat)).to be true
-    expect(@session.resource_collection.include?(keyboard_cat)).to be true
+    recipe.run_context.resource_collection << keyboard_cat
+    # session.rebuild_collection
+    expect(session.resource_collection.include?(kitteh_cat)).to be true
+    expect(session.resource_collection.include?(keyboard_cat)).to be true
   end
 
   it "runs chef with a resource collection from the compiled cookbooks" do
-    allow(@session).to receive(:node_built?).and_return(true)
+    allow(session).to receive(:node_built?).and_return(true)
     chef_runner = double("Chef::Runner.new", converge: :converged)
-    expect(Chef::Runner).to receive(:new).with(an_instance_of(Chef::RunContext)).and_return(chef_runner)
+    expect(Chef::Runner).to receive(:new).with(session.recipe.run_context).and_return(chef_runner)
```

Comment:
This was the first clue as to what was wrong. This was asserting that the argument would be any instance and changing it to be the specific instance in this test failed.